extmod/vfs.c

```diff
@@ -273,7 +273,7 @@ mp_obj_t mp_vfs_umount(mp_obj_t mnt_in) {
         mnt_str = mp_obj_str_get_data(mnt_in, &mnt_len);
     }
     for (mp_vfs_mount_t **vfsp = &MP_STATE_VM(vfs_mount_table); *vfsp != NULL; vfsp = &(*vfsp)->next) {
-        if ((mnt_str != NULL && !memcmp(mnt_str, (*vfsp)->str, mnt_len + 1)) || (*vfsp)->obj == mnt_in) {
+        if ((mnt_str != NULL && !memcmp(mnt_str, (*vfsp)->str, (((*vfsp)->len < mnt_len + 1) ? (*vfsp)->len : mnt_len + 1))) || (*vfsp)->obj == mnt_in) {
```

Comment:
I think a better fix would be to ensure the string lengths are the same, and only do the `memcmp` if they are:
```c
if ((mnt_str != NULL && mnt_len == (*vfsp)->len && memcmp(...)) ...)
```

Reply:
I agree with you! fixed:)